ctk/common/src/main/kotlin/org/codice/compliance/verification/core/SamlDefinedIdentifiersVerifier.kt

```diff
@@ -146,4 +160,52 @@ internal class SamlDefinedIdentifiersVerifier(val node: Node) {
             }
         }
     }
+
+    /** 8.3.7 Persistent Identifier */
+    private fun verifyPersistentIdentifiers() {
+        node.recursiveChildren()
+            .filter { it.attributeText(FORMAT) == PERSISTENT_ID }
+            .forEach {
+                if (it.textContent != null && it.textContent.length > ID_VALUE_LENGTH_LIMIT)
+                    throw SAMLComplianceException.create(SAMLCore_8_3_7_a,
+                        message = "The length of the Persistent ID's value " +
+                            "[${it.textContent.length}] was greater than $ID_VALUE_LENGTH_LIMIT " +
+                            "characters.",
+                        node = it)
+
+                it.attributeText(SP_NAME_QUALIFIER)?.let { nameQualifier ->
+                    if (nameQualifier != idpMetadata.entityId)
+                        throw SAMLComplianceException.create(SAMLCore_8_3_7_b,
+                            SAMLCore_8_3_7_c,
+                            message = "The Persistent ID's NameQualifier [$nameQualifier] is not " +
+                                "equal to ${idpMetadata.entityId}",
+                            node = it)
```

Comment:
When I was looking over my comments, I realized my code was incorrect because it was still passing in "it" for the node which wasn't the node anymore. But you got it 💯